extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -118,13 +120,26 @@ public PolarisEclipseLinkMetaStoreSessionImpl(
   }
 
   /** Load the persistence unit properties from a given configuration file */
-  private Map<String, String> loadProperties(String confFile, String persistenceUnitName) {
+  private Map<String, String> loadProperties(
+      @NotNull String confFile, @NotNull String persistenceUnitName) {
     if (this.properties != null) {
       return this.properties;
     }
 
     try {
-      InputStream input = this.getClass().getClassLoader().getResourceAsStream(confFile);
+      InputStream input = null;
+      int splitPosition = confFile.indexOf("!/");
+      if (splitPosition == -1) {
+        input = this.getClass().getClassLoader().getResourceAsStream(confFile);
```

Comment:
Typically, I use `Thread.currentThread().contextClassLoader()` so that if the plugin jar happens to be loaded by a child classloader, rather than this one, we can still find it.

Reply:
Yes. I made the change to create a new ClassLoader with the conf jar and I'm using Thread.currentThread().contextClassLoader() now.